app/services/scholars_archive/caching_service.rb

```diff
@@ -0,0 +1,21 @@
+require 'net/http'
+
+module ScholarsArchive
+  class CachingService
+    def self.fetch_or_store_in_cache(vocabulary)
+      Rails.cache.fetch("/caching/#{vocabulary}", expires_in: 1.day, race_condition_ttl: 15.seconds) do
+        get_jsonld(vocabulary)
```

Comment:
...also, the ".jsonld" should be coming in from the configuration.. this class would benefit from expecting full URIs and no hard coding.